### PR TITLE
Only read J9Method.extra once

### PR DESCRIPTION
**Pull request by dsouzai**
Since `J9Method.extra` was changed to a volatile in https://github.com/eclipse/openj9/pull/10672, update code that reads the extra field more than once, as suggested in https://github.com/eclipse/openj9/pull/10672#issuecomment-699062420.

### CODE REVIEW

**Comment by dsouzai** [comment]
@DanHeidinga could you please review?

File: runtime/codert_vm/decomp.cpp
Comment (DanHeidinga) [inline runtime/codert_vm/decomp.cpp:262]: Note, this method is called while holding exclusiveVMAccess so it shouldn't matter whether the `extra` field is reread or not.

File: runtime/codert_vm/decomp.cpp
Comment (DanHeidinga) [inline runtime/codert_vm/decomp.cpp:1579]: Similarly, this method is also called under exclusive

File: runtime/vm/createramclass.cpp
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp:1132]: We don't need to introduce a new local for `extra` here.  Better to reuse `frameBuilder`:
```suggestion
	UDATA frameBuilder = (UDATA)currentMethod->extra;
	if ((frameBuilder & J9_STARTPC_NOT_TRANSLATED) != J9_STARTPC_NOT_TRANSLATED) {
```

Note also that this is during class creation so the ramClass hasn't "escaped" to anyone else who can modify it yet

**Review by DanHeidinga — COMMENTED** [review]
minor nitpick but over all this looks OK.

Note, that most of these changes aren't required as the access occur under exclusive but they help set the principle in the code base that we privatize `extra` before using it.

File: runtime/codert_vm/decomp.cpp
Comment (dsouzai) [reply runtime/codert_vm/decomp.cpp:262]: I don't believe the comp thread holds exclusive vm access when it updates the start PC, so even if this thread holds exclusive VM Access, the compilation thread could still update the start PC when it calls `jitMethodTranslated`.

Comment (DanHeidinga) [reply runtime/codert_vm/decomp.cpp:262]: I've been looking through the JIT code and I don't see it acquire / release vm access which is surprising to me.  I wouldn't expect the JIT to get exclusive, but I would have expected "regular" shared vmaccess

Comment (DanHeidinga) [reply runtime/codert_vm/decomp.cpp:262]: @gacholio do you know what the contract for `jitMethodTranslated` is - whether it needs to hold vmaccess or not?

Comment (DanHeidinga) [reply runtime/codert_vm/decomp.cpp:262]: Found it:
https://github.com/eclipse/openj9/blob/05119f696f85cc2137d451e4b6001b94c929b29a/runtime/compiler/control/CompilationThread.cpp#L10052-L10056

I haven't dug back far enough to see the acquire vm access call but the comment indicates this should only occur under vmaccess

Comment (gacholio) [reply runtime/codert_vm/decomp.cpp:262]: Yes, that's right.

Comment (dsouzai) [reply runtime/codert_vm/decomp.cpp:262]: Looks like there's a comment at the top of `compilationEnd` that states that vmaccess must be acquired. It looks like this happens quite high up, so I guess at this point in the code, we must have vmaccess in hand.

Comment (DanHeidinga) [reply runtime/codert_vm/decomp.cpp]: Want to put these changes back in?  My comment at the top of the review was:
> Note, that most of these changes aren't required as the access occur under exclusive but they help set the principle in the code base that we privatize extra before using it.

I **like** the principle being set that we always privatize use of `extra` so I'd prefer to keep these changes.

**Comment by gacholio** [comment]
Please let me know when you think the changes are ready - I'd like to have a last look before they get committed, particularly at the j2iTransition code.

File: runtime/codert_vm/decomp.cpp
Comment (dsouzai) [reply runtime/codert_vm/decomp.cpp]: Ah I see; got confused because the fact about exclusive vmaccess was explicitly noted.

**Comment by dsouzai** [comment]
@gacholio changes are ready.

**Review by DanHeidinga — APPROVED** [review]
lgtm

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,win,osx jdk8

**Comment by DanHeidinga** [comment]
Launching some testing but will wait for @gacholio's review before merging

**Review by gacholio — APPROVED** [review]
I've taken a look at the uses in transition, and they all appear to be correct as-is.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,osx jdk8

**Comment by DanHeidinga** [comment]
Relaunched the osx and plinux builds as the failures appear to be jenkins infra-related, not due to this PR

**Comment by DanHeidinga** [comment]
Build failures are due to UNB artifactory being down

**Comment by gacholio** [comment]
Windows passed, so I would call this ready to merge.

**Comment by DanHeidinga** [comment]
I was going to wait but as there's no ETA on when it'll be back up - let's merge it

**Pull request merged**
